tests/lib/edge_impulse/src/main.cpp

```diff
@@ -434,11 +446,16 @@ ZTEST(suite0, test_data_isr)
 static void setup_fn(void *unused)
 {
 	ARG_UNUSED(unused);
+
 	bool cancelled;
 	int err = ei_wrapper_clear_data(&cancelled);
+	prediction_idx = 0;
+	ei_run_classifier_mock_init();
 
 	zassert_false(cancelled, "Prediction was not cancelled");
 	zassert_ok(err, "Cannot clear data");
+	err = k_sem_take(&test_sem, EI_TEST_SEM_TIMEOUT);
```

Comment:
This can only act as an optional cleanup action. In most cases semaphore would not be available here, so you are expected to get `-EAGAIN` (spurious semaphore give should not happen unless the previous test does something fishy). I would also shorten the timeout a bit (I would suggest e.g. 20 ms to avoid blocking test for too long).